### PR TITLE
Allow adjusting run dates if admin or new run

### PR DESCRIPTION
**Pull request by apmasell**
JIRA: GLT-1793

### CODE REVIEW

File: miso-web/src/main/webapp/pages/editRun.jsp [hunk 1]
```diff
@@ -256,9 +256,22 @@
...
+              </script>
+            </c:when>
+            <c:otherwise>
+              <td id="startDate">
+                <fmt:formatDate pattern="dd/MM/yyyy" value="${run.startDate}"/>
+              </td>
+            </c:otherwise>
+          </c:choose>
+          <c:choose>
+          <c:when test="${(run.health.isDone() and empty run.completionDate) or run.id == 0 or fn:contains(SPRING_SECURITY_CONTEXT.authentication.principal.authorities,'ROLE_ADMIN')}">
```
Comment (callunity) [inline miso-web/src/main/webapp/pages/editRun.jsp]: I don't know that completion date should be editable (even by admins) before the run is actually complete. What do you think?

Comment (djcooke) [inline miso-web/src/main/webapp/pages/editRun.jsp]: True... Also as is, if you wanted to mark something completed and add the completionDate at the same time (as would be normal), you would have to
1. change to Complete
2. save
3. set completionDate
4. save again

Comment (callunity) [inline miso-web/src/main/webapp/pages/editRun.jsp]: Hmm, when I deployed to test, the completion date entry box was visible all the time so I don't know that there would have been a requirement to save twice. I think that before the big recent schema changes, there was an event handler to make the completion date box visible when the status is set to Failed or Completed. IMO, that would be the clearest way to implement it.

Comment (apmasell) [inline miso-web/src/main/webapp/pages/editRun.jsp]: So `isDone && (empty || admin)`? That would still require two saves.

Comment (djcooke) [inline miso-web/src/main/webapp/pages/editRun.jsp]: Make input if admin and use JS to control whether it's editable?

Comment (apmasell) [inline miso-web/src/main/webapp/pages/editRun.jsp]: Turns out that logic was there all along...just broken.

**Comment by callunity** [comment]
Could you please make `startDate` a required field (in the `validateRun` function) if that input is present?

**Comment by apmasell** [comment]
Done. And I check completion date too.

**Comment by apmasell** [comment]
Fixed.

**Comment by apmasell** [comment]
Yeah. I'm not a fan, but It Was Like That When I Found It™.

**Pull request merged**